autoload/cm.vim

```diff
@@ -479,6 +480,11 @@ endfunc
 
 
 func! s:check_changes(...)
+
+	if !get(b:,'cm_enable',0)  || &paste!=0 || g:_cm_lock || mode() != 'i'
```

Comment:
This condition is repeated, and not self documenting.

How about exporting it to a `s:should_skip_completion_work()`?